### PR TITLE
 Optimize install custom python modules in sc-machine

### PR DESCRIPTION
**Pull request by KovalM**
Note: depends on #20

### CODE REVIEW

File: scripts/install_ostis.sh [hunk 1]
```diff
@@ -1,69 +1,74 @@
...
-	sed -i '/python/d' ./config.ini.in
-	sed -i '/modules_path/d' ./config.ini.in 
-	echo -e '[python]' >> ./config.ini.in
-	echo 'modules_path = ${SC_MACHINE_ROOT}/sc-kpm/sc-python/services;${SC_MACHINE_ROOT}/../../problem-solver/py/services' >> ./config.ini.in
-	cd ..
-	
-	cd ./scripts
+	if ! grep -q "${PYTHON_PATH}" "${PLATFORM_PATH}/sc-machine/config/config.ini.in";
+	then
+  	PYTHON_PATH_ESCAPED="$(echo "${PYTHON_PATH}" | sed -e 's/[$/]/\\&/g')"
```
Comment (ValikVityaz) [inline scripts/install_ostis.sh]: Why is there a '$' in `'s/[$/]/\\&/g'`?

File: scripts/install_subsystems.sh [hunk 2]
```diff
@@ -1,12 +1,19 @@
...
-cd ../ostis-web-platform/sc-machine/scripts
+cd ${PLATFORM_PATH}/sc-machine/scripts
 ./make_all.sh
 
-cd ../../scripts
+cd ${PLATFORM_PATH}/scripts
 ./build_kb.sh
 
+cd ${WORKING_PATH}
+
```
Comment (ValikVityaz) [inline scripts/install_subsystems.sh:18]:
```suggestion
```

File: scripts/install_ostis.sh [hunk 3]
```diff
@@ -1,69 +1,74 @@
...
+  	sed -i "/modules_path/ s/$/;${PYTHON_PATH_ESCAPED}/" "${PLATFORM_PATH}/sc-machine/config/config.ini.in"
```
Comment (ValikVityaz) [inline scripts/install_ostis.sh]:
```suggestion
		PYTHON_PATH_ESCAPED="$(echo "${PYTHON_PATH}" | sed -e 's/[$/]/\\&/g')"
		sed -i "/modules_path/ s/$/;${PYTHON_PATH_ESCAPED}/" "${PLATFORM_PATH}/sc-machine/config/config.ini.in"
```

File: scripts/install_ostis.sh [hunk 4]
```diff
@@ -1,69 +1,74 @@
...
+	fi
+
+	cd scripts
 	./make_all.sh
-	cat ../bin/config.ini >> ../../config/sc-web.ini
-	cd ../..
+	cat ../bin/config.ini >> "${PLATFORM_PATH}"/config/sc-web.ini
```
Comment (ValikVityaz) [inline scripts/install_ostis.sh:50]: '..' is replaced everywhere but here. Shouldn't it be `cat "${PLATFORM_PATH}"/sc-machine/bin/config.ini`?

File: scripts/clone_subsystems.sh [hunk 5]
```diff
@@ -1,11 +1,15 @@
...
 
-cd ..
+APP_ROOT_PATH=$(cd "$( dirname "${BASH_SOURCE[0]}" )" >/dev/null 2>&1 && cd .. && pwd)
+PLATFORM_PATH="${APP_ROOT_PATH}/ostis-web-platform"
+WORKING_PATH=$(pwd)
+
+cd ${APP_ROOT_PATH}
 if [ ! -d "./subsystems" ]
   then
     mkdir subsystems
```
Comment (ValikVityaz) [inline scripts/clone_subsystems.sh:11]:
```suggestion
then
	mkdir subsystems
```

File: scripts/clone_subsystems.sh [hunk 6]
```diff
@@ -1,11 +1,15 @@
...
-    echo "../subsystems" >> ./ostis-web-platform/repo.path
+    echo "../subsystems" >> "${PLATFORM_PATH}"/repo.path
```
Comment (ValikVityaz) [inline scripts/clone_subsystems.sh:12]:
```suggestion
	echo "../subsystems" >> "${PLATFORM_PATH}"/repo.path
```

**Review by ValikVityaz — CHANGES_REQUESTED** [review]

File: scripts/install_ostis.sh [hunk 7]
```diff
@@ -1,69 +1,74 @@
 #!/bin/bash
 
+APP_ROOT_PATH=$(cd "$( dirname "${BASH_SOURCE[0]}" )" >/dev/null 2>&1 && cd .. && pwd)
+PLATFORM_PATH="${APP_ROOT_PATH}/ostis-web-platform"
+WORKING_PATH=$(pwd)
+PYTHON_PATH=${APP_ROOT_PATH}/problem-solver/py/services
```
Comment (ValikVityaz) [inline scripts/install_ostis.sh]:
```suggestion
PYTHON_PATH="${APP_ROOT_PATH}"/problem-solver/py/services
```
Should be wrapped in quotes everywhere.

File: scripts/install_ostis.sh (hunk 1 above)
Comment (KovalM) [reply scripts/install_ostis.sh]: This one of symbols that should be escaped

File: scripts/install_ostis.sh [hunk 8]
```diff
@@ -1,69 +1,73 @@
...
-	sed -i '/modules_path/d' ./config.ini.in 
-	echo -e '[python]' >> ./config.ini.in
-	echo 'modules_path = ${SC_MACHINE_ROOT}/sc-kpm/sc-python/services;${SC_MACHINE_ROOT}/../../problem-solver/py/services' >> ./config.ini.in
-	cd ..
+	cat "${APP_ROOT_PATH}"/scripts/sc_machine_cmake_file_ext.txt >> ./CMakeLists.txt
 	
-	cd ./scripts
+	if ! grep -q "${PYTHON_PATH}" "${PLATFORM_PATH}/sc-machine/config/config.ini.in";
+	then
+		PYTHON_PATH_ESCAPED="$(echo "${PYTHON_PATH}" | sed -e 's/[$/]/\\&/g')"
```
Comment (KovalM) [inline scripts/install_ostis.sh]:
```suggestion
		PYTHON_PATH_ESCAPED="$(echo "${PYTHON_PATH}" | sed -e 's/[/]/\\&/g')"
```

**Review by ValikVityaz — APPROVED** [review]

**Review by MikhailSadovsky — APPROVED** [review]

**Pull request merged**